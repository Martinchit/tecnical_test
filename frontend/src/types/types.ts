export interface Stock {
  id: number;
  stockId: string;
  currency: string;
  ric: string;
  bloombergTicker: string;
  bloombergTickerLocal: string;
  name: string;
  country: string;
  price: number;
}

export type OrderSide = 'Buy' | 'Sell';
export type Currency = 'HKD' | 'USD' | 'AUD' | 'EUR' | 'SGD';
export type OrderStatus =
  | 'Not Ready'
  | 'Ready'
  | 'In Progress'
  | 'Booked'
  | 'Rejected';
export type OrderExecutionMode = 'Market' | 'Limit';

export interface StockOrder {
  id: number;
  stockId: string;
  currency: string;
  bloombergTickerLocal: string;
  side: OrderSide;
  status: OrderStatus;
  executionMode?: OrderExecutionMode;
  orderPrice?: number;
  shareAmount?: number;
  error?: string;
  selected: boolean;
  orderId: string;
}

export interface StockListAlert {
  [x: number]: OrderSide;
}

export interface ErrorResponse {
  type: string;
  description: string;
  code: number;
}

export interface SelectedId {
  id: number;
  side: string;
}

export type AuthMode = 'Log In' | 'Sign Up';
