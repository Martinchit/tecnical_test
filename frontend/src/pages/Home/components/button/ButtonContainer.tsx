import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const ButtonContainer = styled(Button)`
  background-color: #fff;
  border: none;
  &:hover {
    outline: none;
    box-shadow: none;
    background-color: #fff;
  }
  &:focus {
    outline: none;
    box-shadow: none;
    background-color: #fff;
  }
`;
