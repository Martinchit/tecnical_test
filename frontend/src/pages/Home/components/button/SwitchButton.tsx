import React from 'react';
import { ButtonContainer } from './ButtonContainer';
import { FlexBox } from '../../../../core/components/atoms/FlexBox';

interface Props {
  authType: string;
  onClick: () => void;
}

export const SwitchButton: React.FC<Props> = ({ authType, onClick }) => (
  <FlexBox margin="10px 0">
    <ButtonContainer id="change_auth_type_button" onClick={onClick} variant="light">
      {authType === 'Log In' ? 'Change to Sign Up' : 'Change to Log In'}
    </ButtonContainer>
  </FlexBox>
);
