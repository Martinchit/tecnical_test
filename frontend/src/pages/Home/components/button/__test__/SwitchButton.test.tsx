import React from 'react';
import { shallow } from 'enzyme';

import { SwitchButton } from '../SwitchButton';
import { ButtonContainer } from '../ButtonContainer';
import { AuthMode } from '../../../../../types/types';

describe('SwitchButton', () => {
  const props = {
    authType: 'Log In' as AuthMode,
    onClick: jest.fn(),
  };

  it('renders correctly', () => {
    const wrapper = shallow(<SwitchButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly in Log In mode', () => {
    const wrapper = shallow(<SwitchButton {...props} />);
    expect(wrapper.find(ButtonContainer).text()).toBe('Change to Sign Up');
  });

  it('renders correctly in Sign Up mode', () => {
    const p = { ...props, authType: 'Sign Up' as AuthMode };
    const wrapper = shallow(<SwitchButton {...p} />);
    expect(wrapper.find(ButtonContainer).text()).toBe('Change to Log In');
  });
});
