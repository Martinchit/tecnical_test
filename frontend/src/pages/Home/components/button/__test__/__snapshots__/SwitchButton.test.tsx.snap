// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton renders correctly 1`] = `
<styled.div
  margin="10px 0"
>
  <Styled(Button)
    id="change_auth_type_button"
    onClick={[MockFunction]}
    variant="light"
  >
    Change to Sign Up
  </Styled(Button)>
</styled.div>
`;
